src/Page.php

```diff
@@ -440,6 +440,63 @@ private function waitForReloadGenerator($eventName, $loaderId)
         }
     }
 
+    /**
+     * Wait until page contains Node.
+     *
+     * @param string $selectors
+     * @param int    $timeout
+     *
+     * @throws Exception\OperationTimedOut
+     *
+     * @return $this
+     */
+    public function waitUntilContains(string $selectors, int $timeout = 60000)
+    {
+        $this->assertNotClosed();
+
+        Utils::tryWithTimeout($timeout * 1000, $this->waitForElement($selectors));
+
+        return $this;
+    }
+
+    /**
+     * @param string $selectors
+     *
+     * @return bool|\Generator
+     *
+     * @internal
+     */
+    public function waitForElement(string $selectors, int $position = 1)
+    {
+        $delay = 500;
+        $element = [];
+
+        while (true) {
+            $this->screenshot()->saveToFile('./public/screen1.png');
+
+            try {
+                $elementList = $this
+                    ->evaluate('JSON.parse(JSON.stringify(document.querySelectorAll("'.$selectors.'")));')
+                    ->getReturnValue();
+
+                $position = \max(0, ($position - 1));
+                $position = \min($position, (\count($elementList) - 1));
+
+                $element = $this
+                    ->evaluate('JSON.parse(JSON.stringify(document.querySelectorAll("'.$selectors.'")['.$position.'].getBoundingClientRect()));')
+                    ->getReturnValue();
```

Comment:
extract?

Reply:
?

Reply:
AFAIK this is snippet is 3 times in the file

introduce getElement()

Reply:
only twice afaik, but i added to utils 🤷